robottelo/hosts.py

```diff
@@ -1662,6 +1662,7 @@ def omit_credentials(self):
         yield
         self.omitting_credentials = False
 
+    @contextmanager
```

Comment:
I wonder why it wasnt a context manager earlier and still worked :octocat:

Reply:
Yes, Session class itself is context manager, so it was working, but to yield its object and for handling user deletion as part teardown or using __exit__(), we need this to be contextmanager as well, I don't see any issue with having this here.
and you mentioned `handle post yield in Session class` , do you mean in airgun, can you elaborate on this ?

Reply:
Forgot the context here, but probably I mean handle that in Session class `__exit__` function as its a context manger.

Reply:
Ping @Gauravtalreja1 !